docs/source/reference-io.rst

```diff
@@ -651,153 +651,110 @@ Spawning subprocesses
+  its outputs and return value. This is what you should reach for if you
+  want to maybe send some input, maybe receive the output (all at once),
+  but don't need incrementality or back-and-forth. It is modelled after
+  the standard :func:`subprocess.run` with some additional features
+  and saner defaults.
+
+* :class:`trio.Process` starts a process in the background and optionally
+  provides Trio streams for interacting with it (sending input,
+  receiving output and errors). Using it requires a bit more code
+  than :func:`~trio.run_process`, but exposes additional capabilities:
+  back-and-forth communication, processing output as soon as it is generated,
+  and so forth. It is modelled after the standard :class:`subprocess.Popen`.
 
 
 .. _subprocess-options:
 
 Options for starting subprocesses
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The standard :mod:`subprocess` module supports a dizzying array
-of `options <https://docs.python.org/3/library/subprocess.html#popen-constructor>`__
-for controlling the environment in which a process starts and the
-mechanisms used for communicating with it. (If you find that list
-overwhelming, you're not alone; you might prefer to start with
-just the `frequently used ones
-<https://docs.python.org/3/library/subprocess.html#frequently-used-arguments>`__.)
-
-Trio makes use of the :mod:`subprocess` module's logic for spawning
-processes, so almost all of these options can be used with their same
-semantics when starting subprocesses under Trio. (You may need to
-``import subprocess`` in order to access constants such as ``PIPE`` or
-``DEVNULL``.)  The exceptions are ``encoding``, ``errors``,
-``universal_newlines`` (and its 3.7+ alias ``text``), and ``bufsize``;
+All of Trio's subprocess APIs accept the numerous keyword arguments used
+by the standard :mod:`subprocess` module to control the environment in
+which a process starts and the mechanisms used for communicating with
+it.  These may be passed wherever you see ``**options`` in the
+documentation below.  See the `full list
+<https://docs.python.org/3/library/subprocess.html#popen-constructor>`__
+or just the `frequently used ones
+<https://docs.python.org/3/library/subprocess.html#frequently-used-arguments>`__
+in the :mod:`subprocess` documentation. (You may need to ``import
+subprocess`` in order to access constants such as ``PIPE`` or
+``DEVNULL``.)
+
 Trio always uses unbuffered byte streams for communicating with a
-process, so these options don't make sense. Text I/O should use a
-layer on top of the raw byte streams, just as it does with sockets.
-[This layer does not yet exist, but is in the works.]
+process, so it does not support the ``encoding``, ``errors``,
+``universal_newlines`` (alias ``text`` in 3.7+), and ``bufsize``
+options. Text I/O should use a layer on top of the raw byte streams,
+just as it does with sockets.  [This layer does not yet exist, but is
+in the works.]
```

Comment:
I'm having some more doubts about including this kind of forward-looking statement in the docs... maybe we should make it just: "Currently, Trio always uses unbuffered byte streams for communicating with a process, so it does not support the encoding, errors, universal_newlines (alias text in 3.7+) and bufsize options." And leave it at that?

I don't know for sure yet whether we will have `TextStream` wrapper objects in trio by default, and if we ever do add them then we might as well implement the arguments here (as a shorthand for wrapping the relevant `Stream`s in `TextStream`s).